app/src/main/kotlin/com/wire/android/util/deeplink/DeepLinkProcessor.kt

```diff
@@ -145,22 +145,32 @@ class DeepLinkProcessor @Inject constructor(
         }
     }
 
+    /**
+     * Format of deeplink to parse: wire://user/domain/user-id
+     */
     private fun getConnectingUserProfile(uri: Uri, switchedAccount: Boolean, accountInfo: AccountInfo.Valid): DeepLinkResult {
-        // todo. handle with domain case, before lastPathSegment. format of deeplink wire://user/domain/user-id
-        return uri.lastPathSegment?.toDefaultQualifiedId(accountInfo.userId.domain)?.let {
-            DeepLinkResult.OpenOtherUserProfile(it, switchedAccount)
-        } ?: return DeepLinkResult.Unknown
+        val segments = uri.pathSegments
+        segments.takeLast(2).let { domainAndUserId ->
```

Comment:
we should first check how many path segmements there is, and if 2 we take the last as the user ID and use self id as the domain and if 3 then take the last 2 as you did here to make sure it will work with older links

Reply:
`takeLast` will take the right amount of elements (after the host, in this case `user`) and in case there are less it will just return the current items as a list, that's why it is the size 2 check inside or defaulting to the old behavior otherwise.

```kotlin
 val url0: Uri = "wire://user".toUri() --> emptyList()
 val url1: Uri = "wire://user/user-id".toUri() --> listOf(user-id)
 val url2: Uri = "wire://user/domain/user-id".toUri() --> listOf(domain, usre-id)
```

I can change it anyway and do it as you suggest.